autogen/agentchat/conversable_agent.py

```diff
@@ -74,25 +78,24 @@ class ConversableAgent(LLMAgent):
     def __init__(
         self,
         name: str,
-        system_message: Optional[Union[str, List]] = "You are a helpful AI Assistant.",
+        system_message: str = "You are a helpful AI Assistant.",
         is_termination_msg: Optional[Callable[[Dict], bool]] = None,
-        max_consecutive_auto_reply: Optional[int] = None,
-        human_input_mode: Optional[str] = "TERMINATE",
-        function_map: Optional[Dict[str, Callable]] = None,
-        code_execution_config: Union[Dict, Literal[False]] = False,
+        max_consecutive_auto_reply: int = MAX_CONSECUTIVE_AUTO_REPLY,
+        human_input_mode: HumanInputMode = "TERMINATE",
+        function_map: Dict[str, Callable] = {},
+        code_execution_config: Union[CodeExecutionConfig, Literal[False]] = False,
```

Comment:
I believe we need to make corresponding changes in docstr as well.

Reply:
We could change the docstring, but the existing docstring is still valid too since it is just a dictionary that allows for type checking